x-pack/plugins/fleet/server/services/actions/actions.ts

```diff
@@ -0,0 +1,252 @@
+      action: actions[i],
+    }));
+  } catch (createBulkActionsError) {
+    throw new FleetActionsError(
+      `Error creating bulk actions: ${createBulkActionsError.message}`,
+      createBulkActionsError
+    );
+  }
+};
+
+export const getActionsByIds = async (
+  esClient: ElasticsearchClient,
+  actionIds: string[]
+): Promise<FleetAction[]> => {
+  try {
+    const getActionsResponse = await esClient.search({
+      index: AGENT_ACTIONS_INDEX,
+      query: {
+        bool: {
+          filter: [
+            {
+              terms: {
+                action_id: actionIds,
+              },
+            },
+          ],
+        },
+      },
+    });
+    return getActionsResponse.hits?.hits?.map((hit) => hit._source ?? {}) as FleetAction[];
+  } catch (getActionsByIdError) {
+    throw new FleetActionsError(
+      `Error getting action: ${getActionsByIdError.message}`,
+      getActionsByIdError
+    );
+  }
+};
+
+export const getActionsWithKuery = async (
+  esClient: ElasticsearchClient,
+  kuery: string
+): Promise<{ actions: FleetAction[]; total: number }> => {
+  try {
+    const query = toElasticsearchQuery(fromKueryExpression(kuery));
+    const getActionSearchResponse = await esClient.search({
+      index: AGENT_ACTIONS_INDEX,
+      query,
+    });
+
+    const actions = getActionSearchResponse.hits.hits.reduce<FleetAction[]>((acc, hit) => {
+      if (hit._source) {
+        acc.push(hit._source as FleetAction);
+      }
+      return acc;
+    }, []);
+
+    return {
+      actions,
+      total: actions.length,
+    };
+  } catch (getActionSearchError) {
+    throw new FleetActionsError(
+      `Error getting actions with kuery: ${getActionSearchError.message}`,
+      getActionSearchError
+    );
+  }
+};
+
+export const getActionResultsByIds = async (
+  esClient: ElasticsearchClient,
+  actionIds: string[]
+): Promise<{ actionsResults: FleetActionResponse[]; total: number }> => {
+  try {
+    const getActionsResultsResponse = await esClient.search({
+      index: AGENT_ACTIONS_RESULTS_INDEX,
+      query: {
+        bool: {
+          filter: [
+            {
+              terms: {
+                action_id: actionIds,
+              },
+            },
+          ],
+        },
+      },
+    });
+    const actionsResults = getActionsResultsResponse.hits.hits.reduce<FleetActionResponse[]>(
+      (acc, hit) => {
+        if (hit._source) {
+          acc.push(hit._source as FleetActionResponse);
+        }
+        return acc;
+      },
+      []
+    );
+
+    return {
+      actionsResults,
+      total: actionsResults.length,
+    };
+  } catch (getActionByIdError) {
+    throw new FleetActionsError(
+      `Error getting action results: ${getActionByIdError.message}`,
+      getActionByIdError
+    );
+  }
+};
+
+export const getActionResultsWithKuery = async (
+  esClient: ElasticsearchClient,
+  kuery: string
+): Promise<{ actionsResults: FleetActionResponse[]; total: number }> => {
```

Comment:
same comment here about usign `items` property for the result

Reply:
7bd6c021b31f6571570bdcef9fcc504be761b1b3